sympy/simplify/simplify.py

```diff
@@ -2540,8 +2540,8 @@ def recurse(arg, **kwargs):
 
     expr = sympify(expr)
 
-    if not isinstance(expr, Basic) or expr.is_Atom or expr in (
-            exp_polar(0), exp_polar(1)):
+    if (not isinstance(expr, Basic) or expr.is_Atom or expr.is_MatrixSymbol
+            or expr in (exp_polar(0), exp_polar(1))):
```

Comment:
here some sort of `isinstance` command would be ok?  or at least consistent with surrounding code.